### PR TITLE
fix(types): reject out-of-range int() and uint() conversions

### PR DESCRIPTION
**Pull request by erikwa**
Currently, `int(double)`, `uint(double)`, `int(uint)`, and `uint(int)` can silently return incorrect values when the input is outside the target type's representable range:

- Conversions from `double` saturate, and `NaN` converts to `0`
- Conversions between `int` and `uint` wrap

This PR adds explicit range checks so that these conversions return an error instead.

Tests cover boundary values and out-of-range inputs that previously produced incorrect results.

This also fixes a few copy-paste errors in the function names included in conversion error messages. Those changes are in a separate commit; I can drop them if you would prefer to keep this PR strictly scoped.

AI assistance disclosure:

- Used Perplexity GPT-5.6 Terra to help troubleshoot the conversion issue and proofread this PR description.
- Used Claude Opus 5 to assist with applying the corresponding `uint.rs` fix to `int.rs`, review the changes, and suggest additional test cases for better test coverage.

I reviewed all generated suggestions before submitting this PR.

### CODE REVIEW

**Comment by alexsnaps** [comment]
This PR fixes a few conformance tests as well, do you feel you can address these? 
They need removing from the `./conformance/src/bin/ignored.txt` file, and regenerate the test suite (do a `cargo fmt` as well, as the generated code is a bit messy).

If not, I can take this over. 

```
    conversions::int::double_int_max_range
    conversions::int::double_int_min_range
    conversions::int::double_range
    conversions::int::uint_range
    conversions::uint::double_range_beyond_uint
    conversions::uint::double_uint_max_range
    conversions::uint::int_neg
```

File: cel/src/common/types/uint.rs [hunk 1]
```diff
@@ -252,19 +252,43 @@ fn uint<'a>(args: Vec<Cow<'a, dyn Val>>) -> Result<Cow<'a, dyn Val>, ExecutionEr
...
     let ret: Result<Box<UInt>, Box<dyn Val>> = match arg.get_type().kind() {
         Kind::UInt => arg.downcast::<UInt>(),
-        Kind::Int => arg
-            .downcast::<CelInt>()
-            .map(|arg| Box::new(UInt::from(*arg.inner() as u64))),
-        Kind::Double => arg
-            .downcast::<CelDouble>()
-            .map(|arg| Box::new(UInt::from(*arg.inner() as u64))),
+        Kind::Int => match arg.downcast::<CelInt>() {
+            Err(arg) => Err(arg),
```
Comment (alexsnaps) [inline cel/src/common/types/uint.rs:256]: Might be able to get rid of these `Err(arg) => Err(arg)` branches, no? Not a big deal tho
I think `.and_then` might replace the `map`?

**Comment by erikwa** [comment]
> This PR fixes a few conformance tests as well, do you feel you can address these? They need removing from the `./conformance/src/bin/ignored.txt` file, and regenerate the test suite (do a `cargo fmt` as well, as the generated code is a bit messy).
> 
> If not, I can take this over.
> 
> ```
>     conversions::int::double_int_max_range
>     conversions::int::double_int_min_range
>     conversions::int::double_range
>     conversions::int::uint_range
>     conversions::uint::double_range_beyond_uint
>     conversions::uint::double_uint_max_range
>     conversions::uint::int_neg
> ```

Sorry, missed running the conformance tests before. I will fix it and incorporate the changes in the same commit as the out-of-range fixes to avoid having it as a separate commit.

**Comment by alexsnaps** [comment]
No worries. Those failures I actually like seeing ;)

File: cel/src/common/types/uint.rs (hunk 1 above)
Comment (erikwa) [reply cel/src/common/types/uint.rs:256]: I looked into using `and_then`, but I do not think it will work without restructuring the error path. A failed `downcast` returns `Box<dyn Val>`, while a failed conversion should return `ExecutionError`. Using `and_then` requires both results to have the same error type, so the outer `match` is probably needed here unless we want to do a bigger restructuring.

Comment (alexsnaps) [reply cel/src/common/types/uint.rs:256]: Ah yeah... I missed the `return` in the nested `Err` branch. We'd need to preemptively in the outer branch `map_err` as we do "just once" after all `match arg.get_type().kind()` branches... One would have to give, I think your choice is indeed the better one!

**Review by alexsnaps — APPROVED** [review]
Thanks a ton for the contribution!

**Comment by erikwa** [comment]
> Thanks a ton for the contribution!

Happy to contribute :)
